### PR TITLE
Support Unicode account names

### PR DESCRIPTION
**Pull request by mbudde**
This pull request includes some changes needed to make the project build and minor cleanup. Feel free to pick and choose, if there are parts you don't want.

### CODE REVIEW

File: src/beancount.pest [hunk 1]
```diff
@@ -38,9 +38,9 @@ valid_non_letter_commodity_char = { QUOTATION_MARK |  "_" | HYPHEN | "." }
 commodity = @{ ASCII_ALPHA_UPPER ~ (ASCII_ALPHA_UPPER | valid_non_letter_commodity_char){0, 22} ~ (ASCII_ALPHA_UPPER | ASCII_DIGIT)? }
 
 //// Account primitives
-account_type = { "Assets" | "Liabilities" | "Income" | "Expenses" | "Equity" }
-account_name_piece = { ":" ~ (ASCII_ALPHA_UPPER | ASCII_DIGIT) ~ (ASCII_ALPHA_LOWER | ASCII_ALPHA_UPPER | ASCII_DIGIT | "-")* }
-account = @{ account_type ~ account_name_piece* }
+account_type = { UPPERCASE_LETTER ~ (LETTER | DECIMAL_NUMBER | "-")* }
```
Comment (twilco) [inline src/beancount.pest]: Huh, I didn't know the account type could be user-defined.  I was working off that assumption based on [this documentation](https://docs.google.com/document/d/1wAMVrKIA2qtRGmoVDSUBJGmYZSygUaR0uOMW1GV3YE0/edit#heading=h.17ry42rqbuiu), but seems like these docs are out-of-date.  Looking at the actual definition of the account confirms you are right - https://bitbucket.org/blais/beancount/src/487cb9b0248427ac6882889b6b8bcc1665ce393e/beancount/core/account.py?at=default&fileviewer=file-view-default#account.py-27.  Good catch!

File: src/beancount.pest [hunk 2]
```diff
@@ -38,9 +38,9 @@ valid_non_letter_commodity_char = { QUOTATION_MARK |  "_" | HYPHEN | "." }
...
+account_name_piece = { ":" ~ (UPPERCASE_LETTER | DECIMAL_NUMBER) ~ (LETTER | DECIMAL_NUMBER | "-")* }
+account = @{ account_type ~ account_name_piece+ }
```
Comment (twilco) [inline src/beancount.pest]: Is an account required to have at least one piece?  If so, TIL.

Comment (mbudde) [reply src/beancount.pest]: Yep, see https://bitbucket.org/blais/beancount/src/487cb9b0248427ac6882889b6b8bcc1665ce393e/beancount/parser/lexer.l?at=default&fileviewer=file-view-default#lexer.l-279 🙂

**Comment by mbudde** [comment]
Unicode characters are only allowed in account names and in quoted strings I believe. The rule for the latter might need to be updated. But other instances of `ASCII_ALPHA_UPPER` et al. should not be changed.

File: src/beancount.pest (hunk 1 above)
Comment (mbudde) [reply src/beancount.pest]: It's documented in the options reference: https://docs.google.com/document/d/1_-T_BvDtUjj9M7liZMNSkrL8pgC60TGMBlYCiV1e4ZM/

**Comment by twilco** [comment]
Great - I've opened #4 to address unicode within quoted strings.  This PR looks good - thanks again!

As a side note, it seems like you have some experience with Beancount.  If you have time and feel inspired to contribute to this project I'd love have ya.  I'll be diligent about creating issues for things that need to get done, and otherwise feel free to just jump in wherever.

**Pull request merged**